### PR TITLE
gh-154523: Fix data-race in `TextIOWrapper.detach()`

### PR DESCRIPTION
**Pull request by brijkapadia**
* Issue: gh-154523

### CODE REVIEW

**Comment by brijkapadia** [comment]
@kumaraditya303, could you take a look whenever you have time?

**Review by kumaraditya303 — APPROVED** [review]
